packages/core/src/Transform.ts

```diff
@@ -717,9 +717,17 @@ export class Transform extends Component {
     }
   }
 
-  private _translate(translation: Vector3, relativeToLocal: boolean = true): void {
+  private _translate(translation: Vector3, relativeToLocal: boolean): void {
     if (relativeToLocal) {
-      this._position.add(translation);
+      const distance = translation.length();
+      if (distance <= MathUtil.zeroTolerance) {
+        return;
+      }
+      const axisLen = translation.transformToVec3(this.localMatrix).length();
```

Comment:
I think this is not `localMatrix` and also not `worldMatrix`, it is `worldMatrix*invert(localMatrix)`

Reply:
Fixed.